test/integration/controller/core/resourceflavor_controller_test.go

```diff
@@ -0,0 +1,155 @@
+			clusterQueue = utiltesting.MakeClusterQueue("foo").
+				Resource(utiltesting.MakeResource("cpu").Flavor(flavor).Obj()).
+				Obj()
+
+			gomega.Expect(k8sClient.Create(ctx, resourceFlavor)).To(gomega.Succeed())
+			gomega.Expect(k8sClient.Create(ctx, clusterQueue)).To(gomega.Succeed())
+		})
+
+		ginkgo.It("Should deleting resourceFlavor when the corresponding clusterQueue updated the resourceFlavor", func() {
+			defer func() { gomega.Expect(framework.DeleteClusterQueue(ctx, k8sClient, clusterQueue)).To(gomega.Succeed()) }()
+
+			ginkgo.By("Delete resourceFlavor")
+			gomega.Expect(framework.DeleteResourceFlavor(ctx, k8sClient, resourceFlavor)).To(gomega.Succeed())
+			var rf kueue.ResourceFlavor
+			gomega.Eventually(func() []string {
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(resourceFlavor), &rf)).To(gomega.Succeed())
+				return rf.GetFinalizers()
+			}, framework.Timeout, framework.Interval).Should(utiltesting.Equal([]string{core.ResourceFlavorFinalizerName}))
+			gomega.Expect(rf.GetDeletionTimestamp()).ShouldNot(gomega.BeNil())
+
+			ginkgo.By("Update clusterQueue's cohort")
+			var cq kueue.ClusterQueue
+			gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(clusterQueue), &cq)).To(gomega.Succeed())
+			cq.Spec.Cohort = "foo-cohort"
+			gomega.Expect(k8sClient.Update(ctx, &cq)).To(gomega.Succeed())
+			gomega.Eventually(func() error {
+				return k8sClient.Get(ctx, client.ObjectKeyFromObject(resourceFlavor), &rf)
+			}, framework.Timeout, framework.Interval).Should(gomega.Succeed())
+
+			ginkgo.By("Update clusterQueue's flavor")
+			gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(clusterQueue), &cq)).To(gomega.Succeed())
+			cq.Spec.Resources[0].Flavors[0].Name = "foo-resourceFlavor"
+			gomega.Expect(k8sClient.Update(ctx, &cq)).To(gomega.Succeed())
+
+			gomega.Eventually(func() bool {
+				err := k8sClient.Get(ctx, client.ObjectKeyFromObject(resourceFlavor), &rf)
+				return err != nil && apierrors.IsNotFound(err)
+			}, framework.Timeout, framework.Interval).Should(gomega.BeTrue())
+		})
+
+		ginkgo.It("Should deleting resourceFlavor when the corresponding clusterQueue deleted", func() {
+			gomega.Expect(framework.DeleteResourceFlavor(ctx, k8sClient, resourceFlavor)).To(gomega.Succeed())
+
+			var rf kueue.ResourceFlavor
+			gomega.Eventually(func() []string {
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(resourceFlavor), &rf)).To(gomega.Succeed())
+				return rf.GetFinalizers()
+			}, framework.Timeout, framework.Interval).Should(utiltesting.Equal([]string{core.ResourceFlavorFinalizerName}))
+			gomega.Expect(rf.GetDeletionTimestamp() != nil).Should(gomega.BeTrue())
+
+			gomega.Expect(framework.DeleteClusterQueue(ctx, k8sClient, clusterQueue)).To(gomega.Succeed())
+			gomega.Eventually(func() bool {
+				err := k8sClient.Get(ctx, client.ObjectKeyFromObject(resourceFlavor), &rf)
+				return err != nil && apierrors.IsNotFound(err)
```

Comment:
could you make this a `GomegaMatcher`? You could have it in `util/testing`

Reply:
done :)